### DISCUSSION TITLE
revive and severity levels

### DISCUSSION THREAD

**Discussion by ealves-pt** [golangci/golangci-lint#1860] [Q&A] [answered]
Hello I've been playing with revive configuration and I was wondering if the severity levels are compatible at all.

For example, I have the following .golangci.yml
```yaml
linters:
  disable-all: true
  enable:
    - revive
 
linters-settings:
  revive:
    severity: error
    rules:
      - name: cognitive-complexity
        arguments: [15]
        severity: warning
      - name: cyclomatic
        arguments: [12]
        severity: warning
```

When running `golanci-lint run  --issues-exit-code 0 --out-format checkstyle > report.xml` the result is:
```xml
<?xml version="1.0" encoding="UTF-8"?>

<checkstyle version="5.0">
  <file name="internal/slack/slack_test.go">
    <error column="1" line="402" message="cognitive-complexity: function okFileUpload has cognitive complexity 17 (&gt; max enabled 15)" severity="error" source="revive">
    </error>
  </file>
</checkstyle>
```

I would expect it to be:
```xml
<?xml version="1.0" encoding="UTF-8"?>

<checkstyle version="5.0">
  <file name="internal/slack/slack_test.go">
    <error column="1" line="402" message="cognitive-complexity: function okFileUpload has cognitive complexity 17 (&gt; max enabled 15)" severity="warning" source="revive">
    </error>
  </file>
</checkstyle>
```

Is this the wrong expectation? I really need to be able to configure the severity levels on a rule level to fine-tune my SonarQube setup.

**Comment by ldez** [comment, marked as answer]
Hello,

I think your problem is related to #1831

The fix will be available in the fix release.

**Reply by ealves-pt** [reply]
Perhaps I did something wrong but I've installed from the master where your PR was merged into but I still have the same result. 😞

**Reply by ealves-pt** [reply]
Actually, I take that back. I did something wrong.

It seems that the top level `revive.severity` overrides any severity level that you might set bellow. So the following worked:

```
linters-settings:
  revive:
    rules:
      - name: cognitive-complexity
        arguments: [15]
        severity: warning
      - name: cyclomatic
        arguments: [12]
        severity: warning
```

Thanks a lot for the hint @ldez 👍
